### PR TITLE
e2e: deflake TestCtlV3AuthGracefulDisable

### PR DESCRIPTION
**Pull request by fuweid**
We should call Wait for grpc-proxy process stop before start. Otherwise, the tcp port won't be released.

Fixes: #14926

Signed-off-by: Wei Fu <fuweid89@gmail.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by ptabor** [comment]
> We should call Wait for grpc-proxy process stop before start. Otherwise, the tcp port won't be released.
> 
> Fixes: #14926

I assume `Close()`. Thank you for fixing this.

**Review by ptabor — APPROVED** [review]

File: tests/framework/e2e/cluster_proxy.go [hunk 1]
```diff
@@ -159,7 +159,13 @@ func (pp *proxyProc) Stop() error {
 	if pp.proc == nil {
 		return nil
 	}
-	if err := pp.proc.Stop(); err != nil && !strings.Contains(err.Error(), "exit status 1") {
+	err := pp.proc.Stop()
+	if err != nil {
+		return err
+	}
+
+	err = pp.proc.Close()
```
Comment (ahrtr) [inline tests/framework/e2e/cluster_proxy.go:167]: Good catch!

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @fuweid

**Comment by ahrtr** [comment]
Retried multiple times, I can always see  failure below,

```
         ctl_v3_move_leader_test.go:59: error closing etcd processes (expect process already closed)
panic: Fail in goroutine after TestCtlV3AuthGracefulDisable has completed

goroutine 558 [running]:
testing.(*common).Fail(0xc000490820)
	/opt/hostedtoolcache/go/1.19.4/x64/src/testing/testing.go:824 +0x1a8
testing.(*common).Errorf(0xc000490820, {0x13ff789, 0x2d}, {0xc00035b6f8, 0x1, 0x1})
	/opt/hostedtoolcache/go/1.19.4/x64/src/testing/testing.go:941 +0xaf
go.etcd.io/etcd/tests/v3/e2e.authGracefulDisableTest({0xc000490820, {0x0, 0x0}, {0x0, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, ...}, ...})
	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_auth_test.go:183 +0x566
go.etcd.io/etcd/tests/v3/e2e.runCtlTest.func2()
	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_test.go:260 +0x11e
created by go.etcd.io/etcd/tests/v3/e2e.runCtlTest
	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_test.go:258 +0x296
FAIL	go.etcd.io/etcd/tests/v3/e2e	87.426s
FAIL
FAIL: 'grpcproxy_e2e' failed at Thu Dec 15 14:52:18 UTC 2022
```

**Comment by fuweid** [comment]
> Retried multiple times, I can always see  failure below,
> 
> 
> 
> ```
> 
>          ctl_v3_move_leader_test.go:59: error closing etcd processes (expect process already closed)
> 
> panic: Fail in goroutine after TestCtlV3AuthGracefulDisable has completed
> 
> 
> 
> goroutine 558 [running]:
> 
> testing.(*common).Fail(0xc000490820)
> 
> 	/opt/hostedtoolcache/go/1.19.4/x64/src/testing/testing.go:824 +0x1a8
> 
> testing.(*common).Errorf(0xc000490820, {0x13ff789, 0x2d}, {0xc00035b6f8, 0x1, 0x1})
> 
> 	/opt/hostedtoolcache/go/1.19.4/x64/src/testing/testing.go:941 +0xaf
> 
> go.etcd.io/etcd/tests/v3/e2e.authGracefulDisableTest({0xc000490820, {0x0, 0x0}, {0x0, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, ...}, ...})
> 
> 	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_auth_test.go:183 +0x566
> 
> go.etcd.io/etcd/tests/v3/e2e.runCtlTest.func2()
> 
> 	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_test.go:260 +0x11e
> 
> created by go.etcd.io/etcd/tests/v3/e2e.runCtlTest
> 
> 	/home/runner/work/etcd/etcd/tests/e2e/ctl_v3_test.go:258 +0x296
> 
> FAIL	go.etcd.io/etcd/tests/v3/e2e	87.426s
> 
> FAIL
> 
> FAIL: 'grpcproxy_e2e' failed at Thu Dec 15 14:52:18 UTC 2022
> 
> ```

checking

**Pull request merged**